crucible-cli/app/Main.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE ImplicitParams #-}
+
+module Main (main) where
+
+import Lang.Crucible.Simulator.ExecutionTree (emptyExtensionImpl)
+
+import Lang.Crucible.Syntax.Concrete (defaultParserHooks)
+import Lang.Crucible.Syntax.Overrides (setupOverrides)
+
+import Lang.Crucible.CLI (SimulateProgramHooks(..), defaultSimulateProgramHooks)
+import qualified Lang.Crucible.CLI.Options as Opts
+
+main :: IO ()
+main =
+  do let ?parserHooks = defaultParserHooks
+     Opts.main "crucible" (\_ -> emptyExtensionImpl) simulationHooks
```

Comment:
Should `"crucible"` be `"crucible-cli"` here?

Reply:
I actually kind of like it better as `crucible`, but I could be convinced otherwise. I think it's awkward to name a CLI program `*-cli`: the fact that I'm typing its name into my command prompt implies that it's a CLI. Additionally, this is the only binary that interfaces directly with Crucible, so it seems apt to name it `crucible`.

Reply:
Note that the executable stanza in the Cabal file matches this naming scheme: 

https://github.com/GaloisInc/crucible/pull/1123/files#diff-b719c603b6f86ffa97657880ea5a3e8a4335c480718149b707e8136b0a560172R137